src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/ClusterGrainDirectory.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+using Orleans.SystemTargetInterfaces;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    internal class ClusterGrainDirectory : SystemTarget, IClusterGrainDirectory
```

Comment:
Maybe a quick summary would help explain the purpose of this class. E.g. that this is an entry point for other clusters to call into. Silos from the local cluster never call it, correct? Except for `UnregisterAfterNonexistingActivation`?